lib/common/xml.c

```diff
@@ -1071,18 +1071,15 @@ pcmk__xml_needs_escape(const char *text, bool escape_quote)
             case '<':
                 return true;
             case '>':
-                // Not necessary, but for symmetry with '<'
                 return true;
             case '&':
                 return true;
             case '"':
-                if (escape_quote) {
-                    return true;
-                }
-                break;
             case '\n':
             case '\t':
-                // Don't escape newline or tab
+                if (for_attr) {
```

Comment:
However, as I think about this more, I think it makes sense for newlines and tabs to be escaped in text nodes too. The main counterpoint was that it makes the XML harder to read. That's true, but as we said back then, conformant XML parsers will convert the newlines and tabs to single spaces, thus changing the output.

XML isn't meant for human eyes except for while we're developing. Any client tool should be passing it to an XML parser. That parser will then convert a `&#x0A;` to a newline, which will show up as a real newline if the parser then displays the text.

Reply:
Never mind. An XML processor replaces newlines and tabs with spaces only during attribute-value normalization. In other words, this does NOT happen to text nodes. Literal newlines and tabs are the same as escaped ones.

Reply:
Looking at all this again has also made me think 714b4204 was unnecessary -- XML isn't really for human eyes -- but it should be fine, as non-ASCII characters aren't "special" characters for an XML processor.